docs/With_Certificate_Design.md

```diff
@@ -0,0 +1,188 @@
+# Draft Spec: `WithCertificate` API – Bearer, mTLS, Dynamic Selection
+
+## 1. Goals
+
+- Keep existing `WithCertificate(X509Certificate2, bool sendX5C)` behavior **unchanged**.
+- Support:
+  - Standard **Bearer** client‑credential tokens.
+  - **mTLS Bearer** (certificate used only for TLS binding).
+  - **mTLS PoP** (existing `.WithMtlsProofOfPossession()` flow).
+- Add a **delegate‑based** `WithCertificate` that can select a certificate **per request** using `IAppConfig`.
+
+## 2. New types
+
+### 2.1 `ClientCertificateUsage`
+
+```csharp
+public enum ClientCertificateUsage
```

Comment:
THere is no need for this? It should be derived from the context?

Reply:
Are you suggesting we add a a request‑level WithMutualTls() to opt into Bearer over mTLS?